### PR TITLE
Remove-is-ie

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

remove `isIE()` helper and its usages

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/core/src/events/TapEvent.ts [hunk 1]
```diff
@@ -72,7 +70,9 @@ export class TapEvent extends Event {
  * @return true if event was generated by pressing `space` or `enter` on a button
  */
 const isButtonEnterOrSpace = (event: MouseEvent | PointerEvent): boolean => {
-  return isIE && event instanceof PointerEvent ? false : event.detail === 0;
+  // check for click, dbclick, mousedown, mouseup events
+  // as described in https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail
+  return event.detail === 0;
```
Comment (Sakchai-Refinitiv) [inline packages/core/src/events/TapEvent.ts]: This condition is quite simple, I think we should remove this function and use `event.detail === 0` directly with code comment.

File: packages/core/src/events/TapEvent.ts [hunk 2]
```diff
@@ -1,7 +1,5 @@
 import { global } from '../utils/global.js';
 
-const isIE = (/Trident/g).test(navigator.userAgent);
```
Comment (Sakchai-Refinitiv) [inline packages/core/src/events/TapEvent.ts:3]: We should find the same code by searching keyword `Trident` in the repository and remove all.

File: packages/core/src/events/TapEvent.ts (hunk 1 above)
Comment (wattachai-lseg) [reply packages/core/src/events/TapEvent.ts]: Refactored.

File: packages/core/src/events/TapEvent.ts (hunk 2 above)
Comment (wattachai-lseg) [reply packages/core/src/events/TapEvent.ts:3]: "search and destroy" completed. 😄

File: packages/core/src/events/TapEvent.ts [hunk 3]
```diff
@@ -333,7 +322,9 @@ const applyEvent = (target: Global): void => {
    * Use this to fire tap events, if `enter` or `space` was pressed on button
    */
   target.addEventListener('click', (event: MouseEvent | PointerEvent) => {
-    if (isButtonEnterOrSpace(event)) {
+    // check for click, dbclick, mousedown, mouseup events
+    // as described in https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail
+    if (event.detail === 0) {
```
Comment (Sakchai-Refinitiv) [inline packages/core/src/events/TapEvent.ts:327]: The code comment is not correct, it must be "check for events from pressing enter or space buttons"

Comment (wattachai-lseg) [reply packages/core/src/events/TapEvent.ts:327]: comment updated.

**Review by Sakchai-Refinitiv — APPROVED** [review]

**Pull request merged**